api/controllers/Irc_EntryController.js

```diff
@@ -64,31 +67,36 @@ module.exports = {
 
   process_event: function (request_body) {
     if (request_body.operation === 'check in') {
-      return this.saveDetainee(request_body)
-        .then(this.saveEvent.bind(this, request_body));
+      return this.saveEvent(request_body);
     }
     throw new ValidationError('Unknown');
   },
 
-  getPID: (entity) => `${entity.centre}_${entity.person_id}`,
-
-  saveEvent: (request_body, detainee) =>
-    Events.create({
+  saveEvent: request_body => {
+    var event = {
+      person_id: request_body.person_id,
+      cid_id: request_body.cid_id,
+      gender: request_body.gender === 'm' ? 'male' : 'female',
+      nationality: request_body.nationality,
+      centre: request_body.centre,
       operation: request_body.operation,
-      timestamp: request_body.timestamp,
-      detainee: detainee
-    }),
+      timestamp: request_body.timestamp
+    };
+    var id = ModelHelpers.getPid.call(event);
```

Comment:
this is mixed in to the model, so you should be able to do DetaineeEvent.getPid
and also remove the `require(ModelHelpers)`